src/Lobby.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "./App.css";
+import GameHUD from "./GameHUD";
+import { checkLogin } from "./Login";
+import { useParams } from "react-router-dom";
+import io from "socket.io-client";
+
+interface ILobby {}
+
+const Lobby = (props: ILobby) => {
+  const [loggedIn, setLoggedIn] = React.useState<boolean>(false);
+  const [users, setUsers] = React.useState<string[]>([]);
+  const [email, setEmail] = React.useState<string>("");
+  const [socket, setSocket] = React.useState(() => io("/"));
+  const [update, setUpdate] = React.useState<string>("");
+  const { id } = useParams();
+
+  React.useEffect(() => {
+    const asyncOps = async () => {
+      const loginStatus = await checkLogin();
+      if (loginStatus.logged_in) {
+        setLoggedIn(loginStatus.logged_in);
+        socket.emit("join", id, loginStatus.email);
+        setUsers((userArray) => userArray.concat([loginStatus.email]));
+        setEmail(loginStatus.email);
+      }
+    };
+    asyncOps();
+  }, []);
+
+  React.useEffect(() => {
+    if (loggedIn) {
+      socket.emit("ping_room", id);
+
+      socket.on("user_disconnected", (user: string) => {
+        setUsers((users) => {
+          const index = users.indexOf(user.toString());
+          if (index != -1) {
+            users.splice(index, 1);
+          }
+          return users;
+        });
+      });
+
+      socket.on("user_connected", (user: string) => {
+        setUsers((userArray) => userArray.concat([user]));
```

Comment:
Not 100% sure if this is convention or law, but I've always seen setState calls that depend on the current state using the same variable name that was declared in the `React.useState` call (in this case on line 12, with `[users, setUsers]`). You can also concat the `user` directly into `users`, rather than concatenating `[user]` -- not sure why they both result in `user` being added to the array, but `users.concat([user])` reads to me like it's going to append an array with one element to `users`, rather than appending the element itself.

So this would be changed to: `setUsers(users => users.concat(user))`